### PR TITLE
CTCP-3004: Allow EIS stub to act as a proxy for selectively sending messages to EIS, or remaining in stub mode

### PR DESCRIPTION
**Pull request by liandra-giacomine**

### CODE REVIEW

File: it/uk/gov/hmrc/transitmovementseisstub/connectors/EISConnectorSpec.scala [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    "returns a Right of unit if the call is successful" in {
+      server.resetAll()
+
+      stub(OK)
+
+      val hc = HeaderCarrier()
+
+      whenReady(connector.post(source)(hc)) {
+        _.isRight mustBe true
+      }
```
Comment (drnaylor) [inline it/uk/gov/hmrc/transitmovementseisstub/connectors/EISConnectorSpec.scala]: We need to ensure that the stub passes the auth token that it received through to EIS (so, by extension, the config should not need the bearer token specified in it).

So, test the stub has the appropriate `Authorization` header, which should match that in the header carrier's `authorization` field.

File: app/uk/gov/hmrc/transitmovementseisstub/controllers/MessagesController.scala [hunk 2]
```diff
@@ -49,29 +54,54 @@ class MessagesController @Inject() (appConfig: AppConfig, cc: ControllerComponen
   private val UUID_PATTERN         = "^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$".r
   private val BEARER_TOKEN_PATTERN = "^Bearer (\\S+)$".r
 
-  def post(): Action[Source[ByteString, _]] = Action(streamFromMemory) {
+  def routeToEIScheck(client: String, customsOffice: String): Boolean = appConfig.clientAllowList.contains(
+    client
+  ) && appConfig.enableProxyMode && !customsOffice.isBlank && (customsOffice == "gb" | customsOffice == "xi")
```
Comment (drnaylor) [inline app/uk/gov/hmrc/transitmovementseisstub/controllers/MessagesController.scala]: microoptimisation, `|` -> `||`.

File: app/uk/gov/hmrc/transitmovementseisstub/controllers/MessagesController.scala [hunk 3]
```diff
@@ -49,29 +54,54 @@ class MessagesController @Inject() (appConfig: AppConfig, cc: ControllerComponen
...
+
+  def post(customsOffice: String): Action[Source[ByteString, _]] = Action.async(streamFromMemory) {
```
Comment (drnaylor) [inline app/uk/gov/hmrc/transitmovementseisstub/controllers/MessagesController.scala:61]: Can we send a request without a customs office? If not, we should probably allow it still by making this `customsOffice` an `Option` and having two routes, one with the customs code in the URL, one without, and using this same method for both endpoints.

File: it/uk/gov/hmrc/transitmovementseisstub/connectors/EISConnectorSpec.scala (hunk 1 above)
Comment (liandra-giacomine) [reply it/uk/gov/hmrc/transitmovementseisstub/connectors/EISConnectorSpec.scala]: good point, removed the config and added test to ensure it's passed on from the HeaderCarrier

**Review by drnaylor — APPROVED** [review]
This passes regression tests so we'll get this in -- I don't forsee any issue with this but let's see what happens.

**Pull request merged**